Neos.Neos/Classes/Service/UserService.php

```diff
@@ -89,7 +95,10 @@ public function getPersonalWorkspaceName()
             return null;
         }
 
-        $username = $this->userDomainService->getUsername($currentUser);
+        $username = $this->userDomainService->getUsername(
+            $currentUser,
+            $this->securityContext->getAccount()->getAuthenticationProviderName()
```

Comment:
`Security\Context::getAccount()` just returns the first account that is authenticated. That is not super "deterministic". is it really what we need here?

Reply:
Do you mean the case that a single Neos user has more that one account associated? Handling this is not covered by the PR. Or are there other scenarios with more than one authenticated account? Thanks for your feedback!

Reply:
See [doc comment ](https://github.com/neos/flow-development-collection/blob/master/Neos.Flow/Classes/Security/Context.php#L458-L460)on the function:
> There might be a more currently authenticated account in the remaining tokens. If you need them you'll have to fetch them directly from the tokens.

Many accounts can be authenticated with one request

Reply:
I am not sure if I understand all implications correctly. But on the other hand, the method that retrieves the $currentUser uses the same functionality:
`Domain/Service/UserService::getCurrentUser` calls `$this->securityContext->getAccount()` - so the current user should be related to the same account as in the code fragment in question.
This PR does not cover the problem, that one user could have multiple accounts / different authentication providers. Should be possible, too, but a lot more complex I guess.
What do you think? Thanks for your feedback!

Reply:
> But on the other hand, the method that retrieves the $currentUser uses the same functionality

Good call! I think that, too, should use `$this->securityContext->getAccountByAuthenticationProviderName($this->defaultAuthenticationProviderName)` (or rather allow the $authProviderName to be specified).
I don't understand all the implications of this and maybe it's really a separate concern to fix. I just stumbled upon this line in your PR